src/knapsack/MultipleKnapsackModel.py

```diff
@@ -0,0 +1,146 @@
+      elif child.tag == 'outcome':
+        self.outcome = child.text.strip()
+      elif child.tag == 'choiceValue':
+        self.choiceValue = child.text.strip()
+      elif child.tag == 'knapsack':
+        self.knapsackSet[child.get('ID')] = child.text.strip()
+      elif child.tag == 'map':
+        container.mapping[child.text.strip()] = [child.get('value'),child.get('cost')]
+      elif child.tag == 'variables':
+        variables = [str(var.strip()) for var in child.text.split(",")]
+      else:
+        raise IOError("MultipleKnapsackModel: xml node " + str (child.tag) + " is not allowed")
+
+
+  def initialize(self, container, runInfoDict, inputFiles):
+    """
+      Method to initialize the MultipleKnapsack model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, runInfoDict, dict, dictionary containing all the RunInfo parameters (XML node <RunInfo>)
+      @ In, inputFiles, list, list of input files (if any)
+      @ Out, None
+    """
+    pass
+
+
+  def run(self, container, inputDict):
+    """
+      This method calculates the sum of the chosen element values and check if the capacity constraints
+      for all knapsacks are satisfied
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, inputDict, dict, dictionary of inputs from RAVEN
+    """
+    totalValue = 0.0
+    knapsackSetValues={}
+
+    # knapsackSetValues is a dictionary in the form {knapsackID: knapsackValue}
+    for knapsack in self.knapsackSet.keys():
+      knapsackSetValues[knapsack] = inputDict[self.knapsackSet[knapsack]][0]
+
+    # List of allowed knapsack IDs
+    elementAllowedValues = list(map(float, self.knapsackSet.keys()))
+    # Add 0.0 which implies that the element has not been assigned to any knapsack
+    elementAllowedValues.append(0.0)
+
+    for key in container.mapping:
+      if key in inputDict.keys() and inputDict[key] in elementAllowedValues:
+        if inputDict[key] > 0.0:
+          knapsackChosen = str(int(inputDict[key][0]))
+          testValue = knapsackSetValues[knapsackChosen] - inputDict[container.mapping[key][1]]
+          if testValue >= 0:
+            knapsackSetValues[knapsackChosen] = knapsackSetValues[knapsackChosen] - inputDict[container.mapping[key][1]][0]
+            totalValue = totalValue + inputDict[container.mapping[key][0]]
+          else:
+            knapsackSetValues[knapsackChosen] = knapsackSetValues[knapsackChosen] - inputDict[container.mapping[key][1]][0]
+            totalValue = totalValue - inputDict[container.mapping[key][0]] * self.penaltyFactor
+      else:
+        raise IOError("MultipleKnapsackModel: variable " + str(key) + " is either not found in the set of input variables or its values is not allowed.")
+
+    numberUnsatConstraints = 0.0
+    for knapsack in knapsackSetValues.keys():
+      if knapsackSetValues[knapsack] < 0:
+        numberUnsatConstraints = numberUnsatConstraints + 1.
```

Comment:
This loop can be combined with previous for loop.

Reply:
Good point: merged with previous loop